src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.BlazorWasmCompanionExtension/src/extension.ts

```diff
@@ -0,0 +1,80 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { spawn } from 'child_process';
+import { acquireDotnetInstall } from './acquireDotnetInstall';
+import { getAvailablePort } from "./getAvailablePort";
+
+let launchDebugProxy: vscode.Disposable;
+let killDebugProxy: vscode.Disposable;
+
+export function activate(context: vscode.ExtensionContext) {
+    const outputChannel = vscode.window.createOutputChannel("Blazor WASM Debug Proxy");
+    const pidsByUrl = new Map<string, number>();
+
+    launchDebugProxy = vscode.commands.registerCommand('ms-blazorwasm-companion.launchDebugProxy', async () => {
+        try {
+            const debuggingPort = await getAvailablePort(9222);
+            const debuggingHost = `http://localhost:${debuggingPort}`;
+
+            let dotnet = "dotnet";
+            // The vscode.env.remoteName property is set when connected to
+            // any kind of remote workspace. See
+            // https://code.visualstudio.com/api/advanced-topics/remote-extensions#varying-behaviors-when-running-remotely-or-in-the-codespaces-browser-editor
+            const isRemote = vscode.env.remoteName !== 'undefined';
```

Comment:
Very interesting so this runs in non-Codespaces scenario too?

Reply:
Yep, it'll work in Docker/WSL as well.

Also, we're defaulting to launching the debug proxy independently in all scenarios. Just making sure to install dotnet if the user is not in a fully local scenario.